cli/src/pcluster/api/controllers/cluster_operations_controller.py

```diff
@@ -482,18 +486,44 @@ def _get_creation_failures(cluster_status, cfn_stack):
 
 
 def _get_details(cfn_stack, verbose):
-    return _get_alarms_details(cfn_stack) if verbose else None
+    if verbose:
+        alarms = _get_alarms_details(cfn_stack)
+        metrics = _get_cluster_health_metrics_details(cfn_stack)
+        stats = _get_stats_details(cfn_stack)
+        return [Detail(alarms=alarms, metrics=metrics, stats=stats)]
+    else:
+        return None
 
 
 def _get_alarms_details(cfn_stack):
-    alarms_in_alarm = cfn_stack.get_alarms_in_alarm()
-
-    if not alarms_in_alarm:
+    raw_alarms = cfn_stack.get_alarms_and_states()
+    if not raw_alarms:
         return None
 
-    # convert AlarmDetail instances to Detail instances
-    alarm_details = [
-        Detail(alarm_type=alarm_detail["alarm_type"], alarm_state=alarm_detail["alarm_state"])
-        for alarm_detail in alarms_in_alarm
-    ]
+    alarm_details = []
+    for raw_alarm in raw_alarms:
+        alarm_type = raw_alarm["alarm_type"]
+        alarm_state_value = raw_alarm["alarm_state"]
+
+        # Check if the alarm_state_value is valid for AlarmState
+        if alarm_state_value not in (AlarmState.OK, AlarmState.ALARM, AlarmState.INSUFFICIENT_DATA, AlarmState.UNKNOWN):
```

Comment:
Here I would check if Alarms are in OK, ALARM or INSUFFICIENT_DATA state and fall back to UNKNOWN instead of raise the error here

if alarm_state_value not in (AlarmState.OK, AlarmState.ALARM, AlarmState.INSUFFICIENT_DATA):
            alarm_state_value = AlarmState.UNKNOWN

Reply:
Great point! Fixed.